### ISSUE TITLE
Group CHAT not found

### ISSUE THREAD

**Issue by BananaNosh** [zeromq/pyre#141]
Hey,
I started the example chat application as written in the README.
However, I always get the error message "Group CHAT not found" when sending a message, though the group should have been created and joined.

**Comment by sphaero** [comment]
Works fine here:
```~/src/pyre master* ± PYTHONPATH=. python3 examples/chat.py
using Python native module No module named 'zyre_pyzmq'
<pyre.zsocket.ZSocket object at 0x7f47f18d0730>
<pyre.zsocket.ZSocket object at 0x7f47f18d0730>
<pyre.zsocket.ZSocket object at 0x7f47f18d0730> {<pyre.zsocket.ZSocket object at 0x7f47f18d0730>: 1}
HMMM
HMMM [b'ENTER', b'\x0c}!\xe8\xc5\x08G\x7f\x9e \xe4e\xdb\xdb\xcc\xc5', b'CHAT', b'{"CHAT_Header1":"example header1","CHAT_Header2":"example header2"}', b'tcp://10.2.4.190:34645']
NODE_MSG TYPE: b'ENTER'
NODE_MSG PEER: 0c7d21e8-c508-477f-9e20-e465dbdbccc5
NODE_MSG NAME: b'CHAT'
NODE_MSG HEADERS: {'CHAT_Header1': 'example header1', 'CHAT_Header2': 'example header2'}
key = CHAT_Header1, value = example header1
key = CHAT_Header2, value = example header2
NODE_MSG CONT: [b'tcp://10.2.4.190:34645']
<pyre.zsocket.ZSocket object at 0x7f47f18d0730> {<pyre.zsocket.ZSocket object at 0x7f47f18d0730>: 1}
HMMM
HMMM [b'JOIN', b'\x0c}!\xe8\xc5\x08G\x7f\x9e \xe4e\xdb\xdb\xcc\xc5', b'CHAT', b'CHAT']
NODE_MSG TYPE: b'JOIN'
NODE_MSG PEER: 0c7d21e8-c508-477f-9e20-e465dbdbccc5
NODE_MSG NAME: b'CHAT'
NODE_MSG CONT: [b'CHAT']
hoi
<pyre.zsocket.ZSocket object at 0x7f47f18d0730> {<zmq.sugar.socket.Socket object at 0x7f47f22003f0>: 1}
CHAT_TASK: b'hoi'
```

```~/src/pyre master* 1 ± PYTHONPATH=. python3 examples/chat.py 
using Python native module No module named 'zyre_pyzmq'
<pyre.zsocket.ZSocket object at 0x7fb1a6aea118>
<pyre.zsocket.ZSocket object at 0x7fb1a6aea118>
<pyre.zsocket.ZSocket object at 0x7fb1a6aea118> {<pyre.zsocket.ZSocket object at 0x7fb1a6aea118>: 1}
HMMM
HMMM [b'ENTER', b'/\xcbv\xa7\x8b\xe6F\xcb\xa8\x8c\xad\xd5\x82\xff\xa50', b'CHAT', b'{"CHAT_Header1":"example header1","CHAT_Header2":"example header2"}', b'tcp://10.2.4.190:46217']
NODE_MSG TYPE: b'ENTER'
NODE_MSG PEER: 2fcb76a7-8be6-46cb-a88c-add582ffa530
NODE_MSG NAME: b'CHAT'
NODE_MSG HEADERS: {'CHAT_Header1': 'example header1', 'CHAT_Header2': 'example header2'}
key = CHAT_Header1, value = example header1
key = CHAT_Header2, value = example header2
NODE_MSG CONT: [b'tcp://10.2.4.190:46217']
<pyre.zsocket.ZSocket object at 0x7fb1a6aea118> {<pyre.zsocket.ZSocket object at 0x7fb1a6aea118>: 1}
HMMM
HMMM [b'JOIN', b'/\xcbv\xa7\x8b\xe6F\xcb\xa8\x8c\xad\xd5\x82\xff\xa50', b'CHAT', b'CHAT']
NODE_MSG TYPE: b'JOIN'
NODE_MSG PEER: 2fcb76a7-8be6-46cb-a88c-add582ffa530
NODE_MSG NAME: b'CHAT'
NODE_MSG CONT: [b'CHAT']
<pyre.zsocket.ZSocket object at 0x7fb1a6aea118> {<pyre.zsocket.ZSocket object at 0x7fb1a6aea118>: 1}
HMMM
HMMM [b'SHOUT', b'/\xcbv\xa7\x8b\xe6F\xcb\xa8\x8c\xad\xd5\x82\xff\xa50', b'CHAT', b'CHAT', b'hoi']
NODE_MSG TYPE: b'SHOUT'
NODE_MSG PEER: 2fcb76a7-8be6-46cb-a88c-add582ffa530
NODE_MSG NAME: b'CHAT'
NODE_MSG GROUP: b'CHAT'
NODE_MSG CONT: [b'hoi']
```

**Comment by BananaNosh** [comment]
Hey,
Thanks for your response.
I used the version exactly as stated in the readme.
But my output is:

using Python native module No module named 'zyre_pyzmq'

<pyre.zsocket.ZSocket object at 0x7f15120df250>

<pyre.zsocket.ZSocket object at 0x7f15120df250>

Hi

Group CHAT not found.

<pyre.zsocket.ZSocket object at 0x7f15120df250> {<zmq.sugar.socket.Socket
object at 0x7f1515cd9660>: 1}

CHAT_TASK: b'Hi'

Hello World

<pyre.zsocket.ZSocket object at 0x7f15120df250> {<zmq.sugar.socket.Socket
object at 0x7f1515cd9660>: 1}

CHAT_TASK: b'Hello World'

Group CHAT not found.

Do you have any idea, what could be the problem?
Any port option on my computer or something?

Best,
Jonathan


Am Di., 13. Aug. 2019 um 12:23 Uhr schrieb Arnaud Loonstra <
notifications@github.com>:

**Comment by sphaero** [comment]
no idea, did you start 2 nodes before chatting? what platform are you on? you should see enter events if they see each other.

**Comment by BananaNosh** [comment]
Hi,
I am on Ubuntu 18.04.3.
If I start the second node, there does not happen anything in the first.
At some point some weeks ago, it worked. However I don't know what I should
have changed since then.



Am Di., 13. Aug. 2019 um 15:15 Uhr schrieb Arnaud Loonstra <
notifications@github.com>:

**Comment by BananaNosh** [comment]
I also tested it on another device, with Zorin(also Ubuntu based) and it
did not work either.

Am Di., 13. Aug. 2019 um 15:25 Uhr schrieb Jonathan Loos <
loos.jonathan.martin@gmail.com>:

**Comment by sphaero** [comment]
is this on physical machines or on virtual ones? If they don't see each other is must be something network related. Are you sure you can do broadcasts on these devices? It's hard to guess what's going on on your machines.

You could try the zyre library instead._

**Comment by BananaNosh** [comment]
They are physical machines.
Thanks for your help so far. I will check if I can find out if there is something a problem within my network.

**Comment by BananaNosh** [comment]
But for using zyre I would have to use C, right?

**Comment by sphaero** [comment]
Yes, but it has bindings to python: https://github.com/zeromq/zyre/tree/master/bindings/python

**Comment by BananaNosh** [comment]
Hi,
I tried to use the python bindings of zyre, but it did not work either.
Therefore, I wanted to ask which version of python you were using during
your test above?
Thanks in advance

Am Di., 13. Aug. 2019 um 16:04 Uhr schrieb Arnaud Loonstra <
notifications@github.com>:

**Comment by sphaero** [comment]
I was using the latest from the master branch. What was the output of the zyre bindings?

**Comment by tlowrimore** [comment]
I'm experiencing the same issue with Pyre.  I'm starting two nodes on the same machine, but when I send a message to the group, I receive the "Group CHAT not found".  I'm running on Pop_OS 19.04 / Python 3.7.

Cheers!

**Comment by sphaero** [comment]
can you post your output of both nodes?

**Comment by tlowrimore** [comment]
@sphaero Thanks for the quick response.  I'm just getting the initial logging output.  No indication of discovery.  That said, I have to abandon the use of this library due to the use of the LGPL licence.

**Comment by sphaero** [comment]
So it means discovery is failing on your platform. Discovery is done through UDP broadcast. You should see udp broadcast packets on network interface using tcpdump.
for example
```
tcpdump -i all -n port 5670
```

**Comment by tlowrimore** [comment]
@sphaero Well, I feel silly.  Turns out, I have very a restrictive set of UFW rules.  As such, the initial UDP broadcast is blocked.  Adding a rule that allows this to pass solves my issue.  Thank you for your help!

**Comment by BananaNosh** [comment]
Hey, I guess I had the same problem. Can't test it out as I recently
changed my operating system, but at least it is working now.
Thanks for the input.

Am Do., 26. Sept. 2019 um 00:46 Uhr schrieb Tim Lowrimore <
notifications@github.com>:
